kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue

```diff
@@ -132,31 +139,95 @@
     },
     mixins: [commonCoreStrings],
     setup() {
+      const store = getCurrentInstance().proxy.$store;
+      const topicId = store.state.route.params.topic_id;
+
       const {
         sectionSettings$,
         selectFromBookmarks$,
         numberOfSelectedBookmarks$,
-        selectFoldersOrExercises$,
+        //selectFoldersOrExercises$,
         numberOfSelectedResources$,
         numberOfResources$,
       } = enhancedQuizManagementStrings;
 
+      // TODO let's not use text for this
       const viewMoreButtonState = ref('no_more_results');
 
       const { windowIsSmall } = useKResponsiveWindow();
 
-      const { initializeExerciseResources, channels, loading, bookmarks, setCurrentTopicId, contentList } = useExerciseResources();
+      //const { channels, loading, bookmarks, contentList } = useExerciseResources();
+
+      const {
+        hasCheckbox,
+        resources,
+        loading: quizResourcesLoading,
+        fetchQuizResources,
+        fetchMoreQuizResources,
+        hasMore,
+      } = useQuizResources({ topicId });
+
+      const _loading = ref(true);
+
+      const channels = ref([]);
+      const bookmarks = ref([]);
+
+      // Load up the channels
+      const channelBookmarkPromises = [
+        ChannelResource.fetchCollection({
+          params: { has_exercises: true, available: true },
+        }).then(response => {
+          console.log('setting channels', response);
+          channels.value = response.map(chnl => {
+            return {
+              ...chnl,
+              id: chnl.root,
+              title: chnl.name,
+              kind: ContentNodeKinds.CHANNEL,
+              is_leaf: false,
+            };
+          });
+        }),
+        ContentNodeResource.fetchBookmarks({ params: { limit: 25, available: true } }).then(
+          data => {
+            console.log(data); // Do we have a `more` here?
+            bookmarks.value = data.results ? data.results : [];
+          }
+        ),
+      ];
+      Promise.all(channelBookmarkPromises).then(() => {
+        _loading.value = false;
+      });
+
+      const loading = computed(() => {
+        return _loading.value || quizResourcesLoading.value;
+      });
+
+      onMounted(() => {
```

Comment:
I don't think this needs to be wrapped in an `onMounted` hook - nothing in here is interacting with the DOM, so this would be called in the equivalent of the `created` life cycle hook, which is just in the `setup` function.

Should also watch the `topicId` to reload in the case it changes within the component's lifecycle (which it will once we've sorted out the global hook issues).